pkg/controller/job/indexed_job_utils.go

```diff
@@ -201,6 +201,28 @@ func (oi orderedIntervals) has(ix int) bool {
 	return oi[hi].First <= ix
 }
 
+func (oi orderedIntervals) isProperSubsetOf(superIntervals orderedIntervals) (int, bool) {
+	if len(superIntervals) == 0 && len(oi) != 0 {
+		return oi.total(), false
+	}
+
+	var diff int
+	for _, sub := range oi {
+		if sub.First == sub.Last {
```

Comment:
This seems covered by the `for` below.

Reply:
Nice catch! Yes, that's right.

Reply:
This mechanism will be changed with the linear search algorithm.
Let me close this thread.